sys/include/net/loramac.h

```diff
@@ -0,0 +1,715 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_loramac LoRaMAC
+ * @ingroup     net
+ * @brief       LoRaMAC definitions
+ *
+ * @{
+ *
+ * @file
+ * @brief       LoRaMAC header definitions
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef NET_LORAMAC_H
+#define NET_LORAMAC_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    LoRaMAC default parameters
+ * @{
+ */
+/**
+ * @brief   Default device EUI
+ *
+ *          8 bytes key, required for join procedure
+ */
+#ifndef LORAMAC_DEV_EUI_DEFAULT
+#define LORAMAC_DEV_EUI_DEFAULT        { 0x00, 0x00, 0x00, 0x00, \
+                                         0x00, 0x00, 0x00, 0x00 }
```

Comment:
Not sure if this is also true for LoRA, but from experience with IEEE 802.15.4 a random EUI might be better than an all 0 one. Of course, this should be in accordance with the standard, so if certain bits have semantics, they should be set accordingly.

Reply:
Leaving 0 values will prevent the node from the joining a LoRaWAN network in any case. That's why by default they use those arrays. For example, the RN2483 module (with the stack already in it) has those values by default.

The arrays are provided by the LoRaWAN provider when one register a new node and they depend on the type of join procedure (either ABP or OTAA).

I can verify what the LoRaWAN specs mention about that to be sure.

Reply:
Sounds convincing enough to me :-)